exchangerates/exchangerates.go

```diff
@@ -0,0 +1,51 @@
+package exchangerates
+
+import (
+	"context"
+	"errors"
+)
+
+// An ExchangeRateSource returns the price of 1 unit of an asset in USD.
+type ExchangeRateSource interface {
+	Last() (float64, error)
+	Start(ctx context.Context)
+}
+
+type averager struct {
+	ignoreOnError bool
+	sources       []ExchangeRateSource
+}
+
+// NewAverager returns an exchange rate source that averages multiple exchange
+// rates.
+func NewAverager(ignoreOnError bool, sources ...ExchangeRateSource) ExchangeRateSource {
+	return &averager{
+		ignoreOnError: ignoreOnError,
+		sources:       sources,
+	}
+}
+
+// Start implements ExchangeRateSource.
+func (a *averager) Start(ctx context.Context) {
+	for i := range a.sources {
+		go a.sources[i].Start(ctx)
+	}
+}
+
+// Last implements ExchangeRateSource.
+func (a *averager) Last() (float64, error) {
+	sum, count := 0.0, 0.0
+	for i := range a.sources {
+		if v, err := a.sources[i].Last(); err == nil {
```

Comment:
We might be missing an edge case here. From the looks of it a source might return `0, nil` when calling `Last` right after creating it. So we need to make sure to avoid that. 

One good precaution would be to ignore `v` when it's `0` cause that should never be a valid result anyway.